packages/dds/migration-shim/src/migrationShim.ts

```diff
@@ -78,6 +87,22 @@ export class MigrationShim extends TypedEventEmitter<IMigrationEvent> implements
 		throw new Error("Method not implemented.");
 	}
 
+	public get target(): LegacySharedTree | ISharedTree {
+		return this.oldTree;
+	}
+
+	public async load(services: IChannelServices): Promise<void> {
+		this._oldTree = (await this.oldFactory.load(
```

Comment:
I am guessing that eventually logic lives here to determine which tree to load?  Or does that logic live in the factory (which then calls loadLegacy vs. loadNew or something)?

No change requested here, just thinking about where that will go.  Maybe add a comment about future intent.

Reply:
It depends on the snapshot in my opinion. We should just use the correct "factory"

Reply:
If the snapshot is a legacy shared tree snapshot, we use the legacy shared tree shim factory. If the snapshot is a new shared tree snapshot, we use the new shared tree shim factory.